recipes-app/mraa/files/0003-iot2050-add-debugfs-pinmux-and-gpio-chardev-support.patch

```diff
@@ -0,0 +1,701 @@
+From 2562134d8e5c3b62dad50bc760a9cc8e5def9eb8 Mon Sep 17 00:00:00 2001
+From: Benedikt Niedermayr <benedikt.niedermayr@siemens.com>
+Date: Thu, 23 Mar 2023 10:21:58 +0100
+Subject: [PATCH 2/2] iot2050: add debugfs pinmux and gpio chardev support
+
+This patch adds support for multiplexing pins via debugfs rather
+than access memory mapped pad-configuration registers.
+The debugfs pinmux offers the possbility to run mraa on iot2050
+platforms as regular user instead of root by adjusting privileges
+on debugfs files.
+
+Bias settings are currently also configured by accessing pinmux.
+Unfortunatelly a proper upstream-like pinconf usage is currently
+not possible.
+
+Signed-off-by: Benedikt Niedermayr <benedikt.niedermayr@siemens.com>
+---
+ src/arm/siemens/iot2050.c | 484 +++++++++++++++++++++++++++++++++++++-
+ 1 file changed, 476 insertions(+), 8 deletions(-)
+
+diff --git a/src/arm/siemens/iot2050.c b/src/arm/siemens/iot2050.c
+index cec25fa..847a2a8 100644
+--- a/src/arm/siemens/iot2050.c
++++ b/src/arm/siemens/iot2050.c
+@@ -24,6 +24,7 @@
+ #include <stdlib.h>
+ #include <string.h>
+ #include <sys/mman.h>
++#include <limits.h>
+ #include <mraa/types.h>
+ 
+ #include "common.h"
+@@ -39,6 +40,9 @@ typedef struct {
+     uint16_t    index;
+     uint16_t    pinmap;
+     int8_t      mode[MAX_MUX_REGISTER_MODE];
++    const char *debugfs_path[MAX_MUX_REGISTER_MODE];
++    const char *pmx_function[MAX_MUX_REGISTER_MODE];
++    const char *pmx_group[MAX_MUX_REGISTER_MODE];
+ }regmux_info_t;
+ 
+ static void *pinmux_instance = NULL;
+@@ -75,6 +79,54 @@ iot2050_get_regmux_by_pinmap(int pinmap)
+     return NULL;
+ }
+ 
++static mraa_result_t
++iot2050_mux_debugfs(const char *base_dir, const char *group, const char *function, const char* gpio_mode)
++{
++    FILE *fd = NULL;
++    char p_pinmux[PATH_MAX];
++    char mux[MRAA_PIN_NAME_SIZE];
++    int ret;
++
++    syslog(LOG_DEBUG, "iot2050: debugfs: enter\n");
++
++    if(!base_dir || !group || !function)
```

Comment:
`if (...)` - `if` is not a function

Reply:
Right. But MRAA seems to not follow that convention. 
Nevertheless, I see that I didn't use that consistently throughout iot2050_mux_debugfs().